lib/schemaUtils.js

```diff
@@ -2271,16 +2271,21 @@ module.exports = {
    * @param {object} components - components defined in the OAS spec. These are used to
    * resolve references while generating params.
    * @param {object} options - a standard list of options that's globally passed around. Check options.js for more.
+   * @param {*} seenRef - References that are repeated. Used to identify circular references.
    * @returns {Object} reference object from the saved components
    * @no-unit-tests
    */
-  getRefObject: function($ref, components, options) {
+  getRefObject: function($ref, components, options, seenRef = {}) {
     var refObj, savedSchema;
 
     if (typeof $ref !== 'string') {
       return { value: `Invalid $ref: ${$ref} was found` };
     }
 
+    if (seenRef[$ref]) {
+      return { value: `<Error: "${$ref}" contains circular references in it.` };
```

Comment:
Minor - maybe an unintended `<` typo before `Error`
```suggestion
      return { value: `Error: "${$ref}" contains circular references in it.` };
```

Reply:
Added change.